src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,44 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        int supplyAmount = 0;
+        int buyAmount = 0;
+
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            String line;
+            while ((line = bufferedReader.readLine()) != null) {
+                String[] parts = line.split(",");
+                if (line.startsWith("supply")) {
+                    supplyAmount += Integer.parseInt(parts[1]);
+                } else {
+                    buyAmount += Integer.parseInt(parts[1]);
+                }
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read data from the file " + fromFileName, e);
+        }
+        writeToFile(toFileName, supplyAmount, buyAmount);
+    }
+
+    public String createReport(int supplyAmount, int buyAmount) {
```

Comment:
details of implementations shouldn't be exposed, choose a different access modifier
```suggestion
    private String createReport(int supplyAmount, int buyAmount) {
```